### DISCUSSION TITLE
Using Pull Dog

### DISCUSSION THREAD

**Discussion by ffMathy** [mastodon/mastodon#27908] [General]
I recently launched a service that I believe could be helpful in this repository. It's free to use (from a demo-cluster pool of free servers), and I need feedback. I think it could really be a win-win for us both.

https://dogger.io

It automatically spawns up a Docker-based test environment for each of your pull requests, based on a docker-compose.yml file that you choose.

I think it could be helpful in general, to test things out in production before they are merged.

What do you think? It is already used by the [Portainer project](https://github.com/portainer/portainer), and they have been very happy with it so far.

**Comment by nightpool** [comment]
we have 1500 open pull requests—how would we manage which ones get test environments available?

**Comment by ffMathy** [comment]
You can configure `lazy` mode, and then just use the `@pull-dog go fetch` command to start up test environments explicitly. Portainer does this too.

I could also add a feature to only make test environments for specific labels - could that be of interest?

Oh, and thank you for showing interest. Very much appreciated.
